components/clock.rst

```diff
@@ -0,0 +1,88 @@
+.. index::
+   single: Clock
+   single: Components; Clock
+
+.. _`clock-component`:
+
+The Clock Component
+===================
+
+The Clock component decouples applications from the system clock. This improves
+testability of time-sensitive logic.
+
+The component provides a ``ClockInterface`` with 3 different clock
+implementations for different use cases.
+
+* ``NativeClock`` provides a way to interact with the system clock.
+* ``MockClock`` is commonly used in tests as a replacement for the
+  ``NativeClock`` to be able to freeze and change the current time using either
+  ``sleep()`` or ``modify()``.
+* ``MonotonicClock`` relies on ``hrtime()`` and provides a high resolution,
+  monotonic clock, when you need a precise stopwatch.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/clock
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+A clock service replaces creating a new ``DateTime`` or
+``DateTimeImmutable``-object for the current time. Instead, you inject the
+``ClockInterface`` and call ``now()``. By default, your application will likely
+use a ``NativeClock``, which always returns the current system time. In tests it is replaced by a ``MockClock``. The
+``MockClock`` is instantiated with a time and does not move forward on its own, meaning when you create an
+instance with a time, it is always kept, unless you change it using
+either ``sleep()`` or ``modify()``. This gives you full control over what your
+code assumes is the current time.
+
+The following example introduces a service utilizing the clock component to
+determine the current time.
+
+    use Symfony\Component\Clock\ClockInterface;
+
+    class ExpirationChecker
+    {
+        public function __construct(
+            private readonly ClockInterface $clock
+        ) {}
+
+        public function isExpired(DateTimeInterface $time): bool
+        {
+            return $this->clock->now() > $time;
+        }
+    }
+
```

Comment:
Could have its own headline IMO

Reply:
I'm sorry, I don't understand? What headline do you have in mind?

Reply:
```suggestion
Testing
-------
```
👍